README.md
# KCFinder web file manager
http://kcfinder.sunhater.com  
Pavel Tzonkov (sunhater@sunhater.com)

## Overview
KCFinder is free open-source replacement of CKFinder web file manager. It can be integrated into FCKeditor, CKEditor, and TinyMCE WYSIWYG web editors (or your custom web applications) to upload and manage images, flash movies, and other files that can be embedded into an editor's generated HTML content.

## Licenses
* GNU General Public License, version 3
* GNU Lesser General Public License, version 3

## Features
* Ajax engine with JSON responses
* Multiple files upload
* Upload files using HTML5 drag and drop from local file manager
* Download multiple files or a folder as single ZIP file
* Clipboard for copying, moving and downloading multiple files
* Easy to integrate and configure in web applications
* Option to select and return several files. For custom applications only
* Auto resize uploaded images. Configurable maximum image resolution
* PNG watermark support
* Configurable thumbnail resolution
* Automaticaly rotate and/or flip uploaded images depending on the orientation info EXIF tag if it exist
* Multiple themes support
* Multilanguage system
* Preview images in full size

## Compatibality
* KCFinder is officialy tested on Apache 2 web server only. Probably it will works on other web servers. If your web server does not support .htaccess files, you should take care turning off script execution in the upload directory.
* PHP 5.3 or better is required. Safe mode should be off.
* At least one of these PHP extensions is required: GD, ImageMagick or GraphicsMagick.
* To work with client-side HTTP cache, the PHP must be installed as Apache module.
* KCFinder supports Fileinfo PHP extension for MIME type detection for uploaded files. If you plan to use this feature, you should to load Fileinfo extension.
* PHP ZIP extension should be loaded in order to have an option to download multiple files and directories as single ZIP file.
* Automatic rotating and flipping images requires PHP EXIF extension.
* Browser side code requires modern browser. KCFinder does not work in Internet Explorer 6, 7 and 8, but you can use Google Chrome Frame ActiveX plugin to get it working.


doc/Changelog
VERSION 3.12 - 2014-07-
-------------------------
* XSS security fix
* Performance fix
* taphold event added. Emulates right-click on touchscreen devices
* Click with Shift key functionality added
* Minor fixes


VERSION 3.11 - 2014-04-21
-------------------------
* "Unknown error." fixes when using `_normalizeFilenames` setting
  and upon new folder creation


VERSION 3.10 - 2014-04-16
-------------------------
* Important secirity fixes


VERSION 3.0 - 2014-04-08
------------------------
* Minor fixes


VERSION 3.0-pre1 - 2014-04-02
-----------------------------
* Now KCFinder requires PHP >= 5.3 becouse of using namespace: "kcfinder"
* Support CSS & JavaScript minifier (on the fly)
* jQuery UI & Uniform support. New theme & theme engine (old themes are not supported)
* Improvements in JavaScript code to be well compressed and faster
* Keep PNG transparency in generated thumbnails
* New image viewer


VERSION 2.54 - 2014-03-12
-------------------------
* Performance fix only


VERSION 2.53 - 2014-02-22
-------------------------
* Session start fix
* TinyMCE 4 support


VERSION 2.52 - 2014-01-20
-------------------------
* Various image drivers support (gd, imagemagick, graphicsmagic)
* Auto-rotate images based on EXIF data
* PNG watermark support


VERSION 2.51 - 2010-08-25
-------------------------
* Drag and drop uploading plugin - big fixes
* Cookies problem when using single words or IPs as hostname resolved
* Vietnamese localization


VERSION 2.5 - 2010-08-23
------------------------
* Drupal module support
* Drag and drop uploading plugin
* Two more language labels
* Localhost cookies bugfix
* Renaming current folder bugfix
* Small bugfixes


VERSION 2.41 - 2010-07-24
-------------------------
* Directory types engine improvement
* New 'denyExtensionRename' config setting added


VERSION 2.4 - 2010-07-20
------------------------
* Online checking if new version is released in About box. To use this
  feature you should to have Curl, HTTP or Socket extension, or
  allow_url_fopen ini setting should be "on"
* New 'denyUpdateCheck' config setting added
* New 'dark' theme added (made by Dark Preacher)
* Additional 'theme' GET parameter to choose a theme from URL
* Thumbnails loading improvement
* Some changes in Oxygen CSS theme
* Replace alert() and confirm() JavaScript functions with good-looking boxes
* Safari 3 right-click fix
* Small bugfixes


VERSION 2.32 - 2010-07-11
-------------------------
* 'filenameChangeChars' and 'dirnameChangeChars' config settings added
* Content-Type header fix for css.php, js_localize.php and
  js/browser/joiner.php
* CKEditorFuncNum with index 0 bugfix
* Session save handler example in core/autoload.php


VERSION 2.31 - 2010-07-01
-------------------------
* Proportional uploaded image resize bugfix
* Slideshow bugfixes
* Other small bugfixes


VERSION 2.3 - 2010-06-25
------------------------
* Replace XML Ajax responses with JSON
* Replace old 'readonly' config option with advanced 'access' option
  PLEASE UPDATE YOUR OLD CONFIG FILE!!!
* Slideshow images in current folder using arrow keys
* Multipe files upload similar to Facebook upload (not works in IE!)
* Option to set protocol, domain and port in 'uploadURL' setting
* Bugfixes


VERSION 2.21 - 2010-11-19
-------------------------
* Bugfixes only


VERSION 2.2 - 2010-07-27
------------------------
* Many bugfixes
* Read-only config option


VERSION 2.1 - 2010-07-04
------------------------
* Endless JavaScript loop on KCFinder disable bugfix
* New config setting whether to generate .htaccess file in upload folder
* Upload to specified folder from CKEditor & FCKeditor direct upload dialog
* Select multiple files bugfixes


VERSION 2.0 - 2010-07-01
------------------------
* Brand new core
* Option to resize files/folders panels with mouse drag
* Select multiple files with Ctrl key
* Return list of files to custom integrating application
* Animated folder tree
* Directory Type specific configuration settings
* Download multiple files or a folder as ZIP file


VERSION 1.7 - 2010-06-17
------------------------
* Maximize toolbar button
* Clipboard for copying and moving multiple files
* Show warning if the browser is not capable to display KCFinder
* Google Chrome Frame support for old versions of Internet Explorer


VERSION 1.6 - 2010-06-02
------------------------
* Support of Windows Apache server
* Support of Fileinfo PHP extension to detect mime types (*mime directory type)
* Option to deny globaly some dangerous extensions like exe, php, pl, cgi etc
* Check for denied file extension on file rename
* Disallow to upload hidden files (with names begins with .)
* Missing last character of filenames without extension bugfix
* Some small bugfixes


VERSION 1.5 - 2010-05-30
------------------------
* Filenames with spaces download bugfix
* FCKEditor direct upload bugfix
* Thumbnail generation bugfixes


VERSION 1.4 - 2010-05-24
------------------------
* Client-side caching bugfix
* Custom integrations - window.KCFinder.callBack()
* Security fixes


VERSION 1.3 - 2010-05-06
------------------------
* Another session bugfix. Now session configuratin works!
* Show filename by default bugfix
* Loading box on top right corner


VERSION 1.2 - 2010-05-03
------------------------
* Thumbnail generation bugfix
* Session bugfix
* other small bugfixes


config.php
<?php

/** This file is part of KCFinder project
  *
  *      @desc Base configuration file
  *   @package KCFinder
  *   @version 3.12
  *    @author Pavel Tzonkov <sunhater@sunhater.com>
  * @copyright 2010-2014 KCFinder Project
  *   @license http://opensource.org/licenses/GPL-3.0 GPLv3
  *   @license http://opensource.org/licenses/LGPL-3.0 LGPLv3
  *      @link http://kcfinder.sunhater.com
  */

/* IMPORTANT!!! Do not comment or remove uncommented settings in this file
   even if you are using session configuration.
   See http://kcfinder.sunhater.com/install for setting descriptions */

$_CONFIG = array(


// GENERAL SETTINGS

    'disabled' => true,
    'uploadURL' => "upload",
    'uploadDir' => "",
    'theme' => "default",

    'types' => array(

    // (F)CKEditor types
        'files'   =>  "",
        'flash'   =>  "swf",
        'images'  =>  "*img",

    // TinyMCE types
        'file'    =>  "",
        'media'   =>  "swf flv avi mpg mpeg qt mov wmv asf rm",
        'image'   =>  "*img",
    ),
    'hideFiles' => array("^\."),

// IMAGE SETTINGS

    'imageDriversPriority' => "imagick gmagick gd",
    'jpegQuality' => 90,
    'thumbsDir' => ".thumbs",

    'maxImageWidth' => 0,
    'maxImageHeight' => 0,

    'thumbWidth' => 100,
    'thumbHeight' => 100,

    'watermark' => "",


// DISABLE / ENABLE SETTINGS

    'denyZipDownload' => false,
    'denyUpdateCheck' => false,
    'denyExtensionRename' => false,


// PERMISSION SETTINGS

    'dirPerms' => 0755,
    'filePerms' => 0644,

    'access' => array(

        'files' => array(
            'upload' => true,
            'delete' => true,
            'copy'   => true,
            'move'   => true,
            'rename' => true
        ),

        'dirs' => array(
            'create' => true,
            'delete' => true,
            'rename' => true
        )
    ),

    'deniedExts' => "exe com msi bat cgi pl php phps phtml php3 php4 php5 php6 py pyc pyo pcgi pcgi3 pcgi4 pcgi5 pchi6",


// MISC SETTINGS

    'filenameChangeChars' => array(/*
        ' ' => "_",
        ':' => "."
    */),

    'dirnameChangeChars' => array(/*
        ' ' => "_",
        ':' => "."
    */),

    'mime_magic' => "",

    'cookieDomain' => "",
    'cookiePath' => "",
    'cookiePrefix' => 'KCFINDER_',


// THE FOLLOWING SETTINGS CANNOT BE OVERRIDED WITH SESSION SETTINGS

    '_normalizeFilenames' => false,
    '_check4htaccess' => true,
    //'_tinyMCEPath' => "/tiny_mce",

    '_sessionVar' => "KCFINDER",
    //'_sessionLifetime' => 30,
    //'_sessionDir' => "/full/directory/path",
    //'_sessionDomain' => ".mysite.com",
    //'_sessionPath' => "/my/path",

    //'_cssMinCmd' => "java -jar /path/to/yuicompressor.jar --type css {file}",
    //'_jsMinCmd' => "java -jar /path/to/yuicompressor.jar --type js {file}",

);

?>
